sites/partners/src/components/listings/PaperListingForm/sections/Units.tsx

```diff
@@ -53,17 +77,34 @@ const FormUnits = ({ units, setUnits, disableUnitsAccordion, featureFlags }: Uni
     }),
   ]
 
-  const nextId = units && units.length > 0 ? units[units.length - 1]?.tempId + 1 : 1
+  const nextId = enableUnitGroups
```

Comment:
I think this is the reason the below issue is happening. This ternary operation looks at `units` when `enableUnitGroups` is true

Reply:
suggestion: for readablity could we remove the double layered ternary operation?
something like 
```
  let nextId = units && units.length > 0 ? units[units.length - 1]?.tempId + 1 : 1

  if (enableUnitGroups) {
    nextId = unitGroups && unitGroups.length > 0 ? unitGroups[unitGroups.length - 1]?.tempId + 1 : 1
  }
```